npm/pkg/dataplane/policies/iptables-comments_linux.go

```diff
@@ -0,0 +1,202 @@
+			- other w/ ns:
+				[!]label1-AND-[!]label2...-AND-ns-[!]labelM-IN-ns-name
+*/
+
+func (networkPolicy *NPMNetworkPolicy) commentForJumpToIngress() string {
+	return networkPolicy.commentForJump(true)
+}
+
+func (networkPolicy *NPMNetworkPolicy) commentForJumpToEgress() string {
+	return networkPolicy.commentForJump(false)
+}
+
+func (networkPolicy *NPMNetworkPolicy) commentForJump(isIngress bool) string {
+	prefix := "EGRESS"
+	if isIngress {
+		prefix = "INGRESS"
+	}
+
+	toFrom := "FROM"
+	if isIngress {
+		toFrom = "TO"
+	}
+
+	podSelectorComment := "all"
+	if len(networkPolicy.PodSelectorList) > 0 {
+		podSelectorComment = commentForInfos(networkPolicy.PodSelectorList)
+	}
+	return fmt.Sprintf("%s-POLICY-%s-%s-%s-IN-ns-%s", prefix, networkPolicy.PolicyKey, toFrom, podSelectorComment, networkPolicy.NameSpace)
+}
+
+func commentForInfos(infos []SetInfo) string {
+	infoComments := make([]string, 0, len(infos))
+	for _, info := range infos {
+		infoComments = append(infoComments, info.comment())
+	}
+	return strings.Join(infoComments, "-AND-")
+}
+
+func (info SetInfo) comment() string {
+	name := info.IPSet.GetPrefixName()
+	if info.Included {
+		return name
+	}
+	return "!" + name
+}
+
+func (aclPolicy *ACLPolicy) comment() string {
+	// cleanPeerList contains peers that aren't namedports
+	var cleanPeerList []SetInfo
+	// TODO remove the if check and initialize the list like in egress once we replace SrcList and DstList with PeerList
+	if aclPolicy.Direction == Ingress {
+		cleanPeerList = aclPolicy.SrcList
+	} else {
+		cleanPeerList = make([]SetInfo, 0, len(aclPolicy.DstList))
+	}
+
+	var namedPortPeer SetInfo
+	foundNamedPortPeer := false
+	for _, info := range aclPolicy.DstList {
+		if info.IPSet.Type == ipsets.NamedPorts {
+			if foundNamedPortPeer {
+				klog.Errorf("while creating ACL comment, unexpectedly found more than one namedPort peer for ACL:\n%s", aclPolicy.String())
+			}
+			namedPortPeer = info
+			foundNamedPortPeer = true
+		} else if aclPolicy.Direction == Egress {
+			// TODO remove the if check once we replace SrcList and DstList with PeerList
+			cleanPeerList = append(cleanPeerList, info)
+		}
+	}
+
+	builder := strings.Builder{}
+	if aclPolicy.Target == Allowed {
+		builder.WriteString("ALLOW")
+	} else {
+		builder.WriteString("DROP")
+	}
+
+	if len(cleanPeerList) == 0 {
+		builder.WriteString("-ALL")
+	} else {
+		if aclPolicy.Direction == Ingress {
+			builder.WriteString("-FROM-")
+		} else {
+			builder.WriteString("-TO-")
+		}
+		builder.WriteString(commentForInfos(cleanPeerList))
+	}
+
+	builder.WriteString(aclPolicy.Protocol.comment())
+	builder.WriteString(aclPolicy.DstPorts.comment())
+	if foundNamedPortPeer {
+		builder.WriteString("-TO-" + namedPortPeer.comment())
+	}
+	return builder.String()
+}
+
+func (proto Protocol) comment() string {
+	if proto == UnspecifiedProtocol {
+		return ""
+	}
+	return fmt.Sprintf("-ON-%s", string(proto))
+}
+
+func (portRange *Ports) comment() string {
```

Comment:
instead of having these  <type>.comment() functions for all in this file, can we place these near their type declarations? that will be cleaner in my opinion ? No strong opinion though

Reply:
I think we should leave as is since comments are only used in Linux, and we don't have a policies_linux.go file

Reply:
@vakalapa is correct, methods should be defined next to their receivers
the farther a declaration is from the usage, the harder it is to grok

Reply:
edit: we do have a policies_linux.go file. Might make sense to compile iptables code into one file?

Reply:
as discussed offline, moving these Linux-specific codes to policy_linux.go